src/lib/ng-package/ngcc-cache.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Registers the paths to package.json files of libraries that have been processed by ngcc. This
+ * cache is reused across all entry-points of a package, so module requests across the entry-points
+ * can determine whether invoking ngcc is necessary.
+ *
+ * The cost of invoking ngcc for an entry-point that has already been processed is limited due to
+ * a fast path in ngcc, however even in this fast-path does ngcc scan the entry-point to determine
+ * if all dependencies have been processed. This cache allows to avoid that work, as entry-points
+ * are processed in batches during which the `node_modules` directory is not mutated.
+ */
+export class NgccProcessingCache {
```

Comment:
Why not simiply use a `Set` directly?

Reply:
I like the documentation aspect of having the dedicated class (JS docblock, method names, etc.). It also provides type-safety with respect to other `Set` instances, so that's nice as well.

Reply:
While you have a point with docs for methods which are not added here :P. 

I don't think it adds anything from a type-safety perspective, both the below are equity type-safe, unless I am missing something.
```ts
const ngccProcessingCache = new Set<string>();
const ngccProcessingCache = new NgccProcessingCache();
```

I don't have a very strong option about this and you can keep it as is. But I do think that in this case using a class doesn't add any value, especially for an internal implementation on top of that which is used in a single place.

Also, while in this case they are not in use, wrapping the Set with a Class has also some disadvantages where you don't expose all the methods and properties available for it. Example, if you need to use `.size` or `entries`, you'd need to implement another method like such https://github.com/ng-packagr/ng-packagr/blob/master/src/lib/file-system/file-cache.ts

Reply:
What I meant by type-safety is the fact that two `Set`s can easily be interchanged, which is not the case with two distinct implementations. I phrased the earlier comment quite poorly, hopefully this makes more sense. It's similar to the `AbsoluteFsPath` branded type that's used in the compiler; although it is a string it does offer identification/documentation and more type-safety compared to just using a string.

> While you have a point with docs for methods which are not added here :P.

😅 I was mainly referring to the docblock on the class itself.

I see what you're saying from an internal implementation perspective and the fact that this thing is used in so few places. I still like the insight in brings when just glancing over the code, and e.g. the ability to ask an IDE where things are marked as processed (which wouldn't be as easy if this were just `Set`).